exercises/Size-biased-extinction.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: exercise
+title: Putting it all together
+subtitle: Size-biased extinction
+language: R
+---
+
+There were a relatively large number of extinctions of mammalian species
+roughly 10,000 years ago. To help understand why these extinctions
+happened scientists are interested in understanding whether there were
+differences in the body size of those species that went extinct and
+those that did not.
+
+To address this question we can use the
+[largest dataset on mammalian body size in the world](http://www.esapubs.org/archive/ecol/E084/094/#data),
+which has data on the mass of recently extinct mammals as well as extant mammals
+(i.e., those that are still alive today). Take a look at the
+[metadata](http://www.esapubs.org/archive/ecol/E084/094/metadata.htm) to
+understand the structure of the data. One key thing to remember is that species
+can occur on more than one continent, and if they do then they will occur more
+than once in this dataset. Also let's ignore species that went extinct in the
+very recent past (designated by the word `"historical"` in the `"status"`
+column).
+
+Import the data into R. If you've looked at a lot of data you'll realize that
+this dataset is tab delimited. Use the argument `sep = "\t"` in `read.csv()` to
+properly format the data. There is no header row, so use `head = FALSE`. The
+unknown value used in the dataset is `-999`. R assumes your unknown value is
+`NA`, but `"NA"` in the data is the code for North America.  Use the additional
+arguments `stringsAsFactors = FALSE, na.strings = "-999"` in `read.csv()` to get
+R to keep `"NA"` as a string and transform `-999` to `NA`.
+
+It's probably a good idea to add column names to help identify columns:
+ 
+```
+colnames(mammal_sizes) <- c("continent", "status", "order", 
+"family", "genus", "species", "log_mass", "combined_mass", 
+"reference")
+```
+
+1. Create a new RStudio project and a new version control repository for this
+   exercise and commit your changes in small logical chunks. If you would like a
+   private repository on GitHub please ask your instructor to set one
+   up. Throughout the assignment focus on using good style to make the code easy
+   for you or someone else to read.
+2. Calculate the mean mass of the extinct species and the mean mass of the
+   extant species. Don't worry about species that occur more than once. We'll
+   consider the values on different continents to represent independent data
+   points.
+3. It looks like the species that went extinct are larger on average, but there
```

Comment:
Assuming students will read through the whole problem before trying to solve it: "It should look like the species..." (Don't feel strongly about this.)